### PR TITLE
Feature: Allow to load a core plugin as external

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Add the possibility to skip loading a core plugin and load it as external instead (behind a feature flag). Example configuration:

```ini
[plugin.testdata]
as_external = true

[feature_toggles]
externalCorePlugins = true
```

**Why do we need this feature?**

Allows to load decoupled core plugins.

**Which issue(s) does this PR fix?**:

Ref https://github.com/grafana/grafana/issues/68973

**Special notes for your reviewer:**

Extracted from https://github.com/grafana/grafana/pull/72205

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 1]
```diff
@@ -2195,6 +2195,12 @@ Available in Grafana v9.5.0 or later, and [OpenTelemetry must be configured as w
 
 If `true`, propagate the tracing context to the plugin backend and enable tracing (if the backend supports it).
 
+## as_external
+
+Load an external versions of a core plugin if it has been installed.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2200]:
```suggestion
Load an external version of a core plugin if it has been installed.
```

**Review by chri2547 — COMMENTED** [review]
One docs suggestions, otherwise looks good! Thank you!

**Review by xnyo — APPROVED** [review]
LGTM!

**Review by sunker — APPROVED** [review]
Really cool!

**Pull request merged**